packages/fxa-settings/src/pages/Signin/Signin/index.stories.tsx

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import Signin from '.';
+import AppLayout from '../../../components/AppLayout';
+import { LocationProvider } from '@reach/router';
+import { Meta } from '@storybook/react';
+import { EXAMPLE_EMAIL, EXAMPLE_SERVICE, EXAMPLE_OTHER_ICON } from './mocks';
+export default {
+  title: 'pages/Signin/SigninPage',
+  component: Signin,
+} as Meta;
+
+// TODO: Add in error and success states when the Alert Bar is implemented
+
+export const PasswordNeeded = () => (
+  <LocationProvider>
+    <AppLayout>
+      <Signin
+        email={EXAMPLE_EMAIL}
+        isPasswordNeeded={true}
+        isPocketClient={false}
+      />
+    </AppLayout>
+  </LocationProvider>
+);
+
+export const PasswordNotNeeded = () => (
+  <LocationProvider>
+    <AppLayout>
+      <Signin
+        email={EXAMPLE_EMAIL}
+        isPasswordNeeded={false}
+        isPocketClient={false}
+      />
+    </AppLayout>
+  </LocationProvider>
+);
+
+export const CustomServiceName = () => (
+  <LocationProvider>
+    <AppLayout>
+      <Signin
+        email={EXAMPLE_EMAIL}
+        isPasswordNeeded={false}
+        isPocketClient={false}
+        serviceName={EXAMPLE_SERVICE}
+      />
+    </AppLayout>
+  </LocationProvider>
+);
+
+export const IsPocketClient = () => (
+  <LocationProvider>
+    <AppLayout>
+      <Signin
+        email={EXAMPLE_EMAIL}
+        isPasswordNeeded={false}
+        isPocketClient={true}
+        serviceName={'Pocket'}
+      />
+    </AppLayout>
+  </LocationProvider>
+);
+
+export const HasServiceLogo = () => (
+  <LocationProvider>
+    <AppLayout>
+      <Signin
+        email={EXAMPLE_EMAIL}
+        isPasswordNeeded={false}
+        isPocketClient={false}
+        serviceName={'VPN'}
+        serviceLogo={EXAMPLE_OTHER_ICON}
+      />
```

Comment:
🤔 It definitely feels like there's some reuse opportunities between these stories and tests. There's a `Subject` pattern we've used in some other places, but might be more appropriate here to just do something like this in the mocks file, if you want to DRY it up:

```
export const SignInWithServiceLogo = () => (
  <Signin
    email={EXAMPLE_EMAIL}
    isPasswordNeeded={false}
    isPocketClient={false}
    serviceName={'VPN'}
    serviceLogo={EXAMPLE_OTHER_ICON}
  />
)
```

Then, we can use it in both files and if we ever need to change logic in the component, it's a bit easier maintenance wise. Could also potentially make a helper function to give you that `LocationProvider` and `AppLayout` nesting which would probably be nice in other storybook files as well, but that's just being more nit picky.